handlers/metrics_test.go

```diff
@@ -173,6 +174,109 @@ func TestMetricsBounties(t *testing.T) {
 		assert.Equal(t, res[0].BountyDescription, "test bounty")
 		assert.Equal(t, res[0].BountyCreated, int64(1112))
 	})
+
+	t.Run("should fetch bounties from db for selected providers", func(t *testing.T) {
+		db.RedisError = errors.New("redis not initialized")
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(mh.MetricsBounties)
+		dateRange := db.PaymentDateRange{
+			StartDate: "1111",
+			EndDate:   "2222",
+			Providers: []string{"provider1", "provider2", "provider3"},
+		}
+		body, _ := json.Marshal(dateRange)
+		req, err := http.NewRequestWithContext(ctx, http.MethodPost, "/boutnies", bytes.NewReader(body))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Provide multiple provider IDs in the request query parameters
+		req.URL.RawQuery = "provider=provider1&provider=provider2&provider=provider3"
+
+		// Mock bounties data for multiple providers
+		bounties := []db.Bounty{
+			{
+				ID:          1,
+				OwnerID:     "provider1",
+				Price:       100,
+				Title:       "bounty 1",
+				Description: "test bounty",
+				Created:     1112,
+			},
+			{
+				ID:          2,
+				OwnerID:     "provider1",
+				Price:       100,
+				Title:       "bounty 2",
+				Description: "test bounty",
+				Created:     1112,
+			},
+			{
+				ID:          3,
+				OwnerID:     "provider2",
+				Price:       200,
+				Title:       "bounty 3",
+				Description: "test bounty",
+				Created:     1113,
+			},
+			{
+				ID:          4,
+				OwnerID:     "provider2",
+				Price:       200,
+				Title:       "bounty 4",
+				Description: "test bounty",
+				Created:     1113,
+			},
+			{
+				ID:          5,
+				OwnerID:     "provider3",
+				Price:       200,
+				Title:       "bounty 5",
+				Description: "test bounty",
+				Created:     1113,
+			},
+			{
+				ID:          6,
+				OwnerID:     "provider3",
+				Price:       200,
+				Title:       "bounty 6",
+				Description: "test bounty",
+				Created:     1113,
+			},
+		}
+		// Mock the database call to return bounties for the selected providers
+		mockDb.On("GetBountiesByDateRange", dateRange, req).Return(bounties).Once()
```

Comment:
You don't need to mock this amount of Providers, the you don't have this amount of DB calls in the request handler

Reply:
Addressed